composer/lib/dependabot/composer/update_checker/version_resolver.rb

```diff
@@ -72,10 +72,7 @@ def fetch_latest_resolvable_version
         def fetch_latest_resolvable_version_string
           base_directory = dependency_files.first.directory
           SharedHelpers.in_a_temporary_directory(base_directory) do
-            File.write("composer.json", prepared_composer_json_content)
-            File.write("composer.lock", lockfile.content) if lockfile
-            File.write("auth.json", auth_json.content) if auth_json
-
+            write_dependency_files
```

Comment:
Super nit but usually named this type of thing elsewhere like so:

```suggestion
            write_temporary_dependency_files
```

Also shows how much duplication there is.. would be great to chat through different approaches to refactoring this at some point. Maybe we can start chipping away if we nail down the approach/philosophy.